### PR TITLE
Fix build and regression

### PR DESCRIPTION
**Pull request by BridgeAR**
This fixes the regression introduced in 2.1.2 and fixes the build script to use iojs 1.0 as long as travis can't install the proper nan version (needed for pg-native).
I also inserted missing changelog entries and fixed a deprecation message in a test.

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -1630,7 +1630,7 @@ module.exports = (function() {
         }
       });
     } else {
-      options.fields = _.intersection(Object.keys(values), Object.keys(this.tableAttributes));
+      options.fields = Object.keys(this.tableAttributes);
```
Comment (mickhansen) [inline lib/model.js]: That's not exactly a proper fix, the old behaviour was very wrong aswell, updatedAt needs to be added to fields like we do in instance.save() i guess.

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,8 @@ language: node_js
 node_js:
   - "0.10"
   - "0.12"
-  - "iojs"
+  - "iojs-v1.0.0"
```
Comment (mickhansen) [inline .travis.yml]: possible to test vs a higher version? 1.6/1.8?

File: lib/model.js (hunk 1 above)
Comment (BridgeAR) [inline lib/model.js]: I agree

**Comment by BridgeAR** [comment]
I also fixed an still outstanding issue with transactions not rolling back if no thenable was provided.

File: .travis.yml [hunk 3]
```diff
@@ -32,7 +33,6 @@ script:
 branches:
   only:
     - master
-    - 1.7.0
```
Comment (mickhansen) [inline .travis.yml]: We're still running tests on 1.7 for the oddball PR.

**Comment by mickhansen** [comment]
Just the last comment and everything looks good.

**Comment by BridgeAR** [comment]
Done

**Pull request merged**